src/main/java/htsjdk/variant/vcf/VCFCompoundHeaderLine.java

```diff
@@ -202,6 +202,9 @@ protected VCFCompoundHeaderLine(String line, VCFHeaderVersion version, Supported
     }
 
     private void validate() {
+        if (countType == VCFHeaderLineCount.INTEGER && count <= 0)
```

Comment:
Does this check conflict with the one on line 213 below that sets `count` to `0`? Maybe this validation should be done after that check?